visualization/app/codeCharta/services/loadInitialFile/loadInitialFile.service.ts

```diff
@@ -25,50 +71,351 @@ export class LoadInitialFileService {
+			case "margin":
+				this.store.dispatch(setMargin({ value }))
+				break
+			default: {
+				const exhaustiveCheck: never = key
+				throw new Error(`Unhandled key: ${exhaustiveCheck}`)
+			}
+		}
+	}
+
+	private mapAppSettingToAction(key: keyof AppSettings, value: any) {
+		switch (key) {
+			case "amountOfTopLabels":
+				this.store.dispatch(setAmountOfTopLabels({ value }))
+				break
+			case "amountOfEdgePreviews":
+				this.store.dispatch(setAmountOfTopLabels({ value }))
+				break
+			case "edgeHeight":
+				this.store.dispatch(setEdgeHeight({ value }))
+				break
+			case "scaling":
+				this.store.dispatch(setScaling({ value }))
+				break
+			case "hideFlatBuildings":
+				this.store.dispatch(setHideFlatBuildings({ value }))
+				break
+			case "invertHeight":
+				this.store.dispatch(setInvertHeight({ value }))
+				break
+			case "invertArea":
+				this.store.dispatch(setInvertArea({ value }))
+				break
+			case "isWhiteBackground":
+				this.store.dispatch(setIsWhiteBackground({ value }))
+				break
+			case "mapColors":
+				this.store.dispatch(setMapColors({ value }))
+				break
+			case "isPresentationMode":
+				this.store.dispatch(setPresentationMode({ value }))
+				break
+			case "showOnlyBuildingsWithEdges":
+				this.store.dispatch(setShowOnlyBuildingsWithEdges({ value }))
+				break
+			case "isEdgeMetricVisible":
+				this.store.dispatch(setIsEdgeMetricVisible({ value }))
+				break
+			case "resetCameraIfNewFileIsLoaded":
+				this.store.dispatch(setResetCameraIfNewFileIsLoaded({ value }))
+				break
+			case "isLoadingMap":
+				this.store.dispatch(setIsLoadingMap({ value }))
+				break
+			case "isLoadingFile":
+				this.store.dispatch(setIsLoadingFile({ value }))
+				break
+			case "sortingOrderAscending":
+			case "isSearchPanelPinned":
+				break
+			case "showMetricLabelNameValue":
+				this.store.dispatch(setShowMetricLabelNameValue({ value }))
+				break
+			case "showMetricLabelNodeName":
+				this.store.dispatch(setShowMetricLabelNodeName({ value }))
+				break
+			case "layoutAlgorithm":
+				this.store.dispatch(setLayoutAlgorithm({ value }))
+				break
+			case "maxTreeMapFiles":
+				this.store.dispatch(setMaxTreeMapFiles({ value }))
+				break
+			case "sharpnessMode":
+				this.store.dispatch(setSharpnessMode({ value }))
+				break
+			case "experimentalFeaturesEnabled":
+				this.store.dispatch(setExperimentalFeaturesEnabled({ value }))
+				break
+			case "screenshotToClipboardEnabled":
+				this.store.dispatch(setScreenshotToClipboardEnabled({ value }))
+				break
+			case "colorLabels":
+				this.store.dispatch(setColorLabels({ value }))
+				break
+			case "isColorMetricLinkedToHeightMetric":
+				this.store.dispatch(setIsColorMetricLinkedToHeightMetricAction({ value }))
+				break
+			case "enableFloorLabels":
+				this.store.dispatch(setEnableFloorLabels({ value }))
+				break
+			default: {
+				const exhaustiveCheck: never = key
+				throw new Error(`Unhandled key: ${exhaustiveCheck}`)
```

Comment:
Throwing errors at this position will most likely prevent to restore old state, right?
Maybe the approach to just ignore unknown actions would be more beneficial for the users.

Reply:
This is actually just a ts-lint "hack" to make sure all possible values are checked in the switch statement. This will not  throw an error during runtime.